addons/website/models/ir_actions.py

```diff
@@ -2,64 +2,60 @@
-        base_url = self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context)
-        link = website_path or xml_id or (id and '%d' % id) or ''
+    @api.multi
+    def _compute_website_url(self):
+        self.ensure_one()
+        base_url = self.env['ir.config_parameter'].get_param('web.base.url')
+        link = self.website_path or self.xml_id or (self.id and '%d' % self.id) or ''
         if base_url and link:
             path = '%s/%s' % ('/website/action', link)
             return '%s' % urlparse.urljoin(base_url, path)
         return ''
 
-    def _get_website_url(self, cr, uid, ids, name, args, context=None):
-        res = dict.fromkeys(ids, False)
-        for action in self.browse(cr, uid, ids, context=context):
+    @api.multi
+    def _get_website_url(self):
+        for action in self:
             if action.state == 'code' and action.website_published:
-                res[action.id] = self._compute_website_url(cr, uid, action.id, action.website_path, action.xml_id, context=context)
-        return res
-
-    _columns = {
-        'xml_id': fields.function(
-            osv.osv.get_xml_id, type='char', string="External ID",
-            help="ID of the action if defined in a XML file"),
-        'website_path': fields.char('Website Path'),
-        'website_url': fields.function(
-            _get_website_url, type='char', string='Website URL',
-            help='The full URL to access the server action through the website.'),
-        'website_published': fields.boolean(
-            'Available on the Website', copy=False,
+                action.website_url = action._compute_website_url()
+
+    xml_id = fields.Char(compute=models.BaseModel.get_xml_id, string="External ID",
+            help="ID of the action if defined in a XML file")
+    website_path = fields.Char('Website Path')
+    website_url = fields.Char(compute='_get_website_url', string='Website URL',
+            help='The full URL to access the server action through the website.')
+    website_published = fields.Boolean('Available on the Website', copy=False,
             help='A code server action can be executed from the website, using a dedicated'
                  'controller. The address is <base>/website/action/<website_path>.'
                  'Set this field as True to allow users to run this action. If it'
-                 'set to is False the action cannot be run through the website.'),
-    }
+                 'set to is False the action cannot be run through the website.')
 
-    def on_change_website_path(self, cr, uid, ids, website_path, xml_id, context=None):
-        values = {
-            'website_url': self._compute_website_url(cr, uid, ids and ids[0] or None, website_path, xml_id, context=context)
-        }
-        return {'value': values}
+    @api.onchange('website_path')
+    def on_change_website_path(self):
+        self.website_url = self._compute_website_url()
```

Comment:
hello @rim-odoo , not needed  so just removed but it's going to change website_url on save,  not instant effect like on_change. Thanks